mtgjson5/providers/cardmarket/set_name_translations.py

```diff
@@ -0,0 +1,118 @@
+import re
+from collections import defaultdict
+from typing import Any, Dict, List, Optional, Tuple, Union
+
+import bs4
+import requests
+import requests_cache
+from singleton_decorator import singleton
+
+from ...classes import MtgjsonTranslationsObject
+from ...utils import LOGGER, retryable_session
+from ..abstract import AbstractProvider
+
+
+@singleton
+class CardMarketProviderSetNameTranslations(AbstractProvider):
+    CARD_MARKET_URL = "https://www.cardmarket.com/{}/Magic/Products/Booster-Boxes?mode=gallery&idCategory=7&idExpansion=0&sortBy=date_asc&site={}"
+    SUPPORTED_LOCALES = {
+        "en": "English",
+        "fr": "French",
+        "de": "German",
+        "it": "Italian",
+        "es": "Spanish",
+    }
+    SESSION: Union[requests.Session, requests_cache.CachedSession]
+    translation_table: Dict[str, Dict[str, str]]
+
+    def __init__(self) -> None:
+        super().__init__({})
+        self.SESSION = retryable_session()
+        self.translation_table = {}
+
+    def _build_http_header(self) -> Dict[str, str]:
+        raise NotImplementedError()
+
+    def download(self, url: str, params: Dict[str, Union[str, int]] = None) -> Any:
+        raise NotImplementedError()
+
+    def get_set_translation_object(self, set_code: str) -> MtgjsonTranslationsObject:
+        if not self.translation_table:
+            self.translation_table = self.__generate_set_name_translations()
+
+        if set_code in self.translation_table.keys():
+            return MtgjsonTranslationsObject(self.translation_table[set_code])
+
+        return MtgjsonTranslationsObject()
+
+    def __generate_set_name_translations(self) -> Dict[str, Dict[str, str]]:
+        translation_table: Dict[str, Dict[str, str]] = defaultdict(
+            lambda: defaultdict(str)
+        )
+
+        for page_number in range(1, 20):
+            for locale_short, locale_long in self.SUPPORTED_LOCALES.items():
+                for position, set_name_localized in self.__download_localized_data(
+                    locale_short, page_number
+                ):
```

Comment:
imo this would be more reable on a single line

Reply:
black formatting unfortunately